src/route-recognizer.js

```diff
@@ -73,7 +85,7 @@ export class RouteRecognizer {
     }
 
     currentState.handlers = handlers;
-    currentState.regex = new RegExp(regex + '$');
+    currentState.regex = caseSensitiveRoute ? new RegExp(regex + '$') : new RegExp(regex + '$', 'i');
```

Comment:
It'd be great to avoid duplicating the ctor call, and simply switch the second parameter out instead.